cardano-cli/src/Cardano/CLI/EraBased/Run/Genesis/CreateTestnetData.hs

```diff
@@ -338,40 +338,41 @@ runGenesisCreateTestNetDataCmd
 
     addDRepsToConwayGenesis
       :: [VerificationKey DRepKey]
+      -- \^ The credential of the DReps
       -> [VerificationKey StakeKey]
+      -- \^ The credentials of those that delegate their votes to the dreps. It happens
+      -- to only be stake keys now, but it could be more general in the future.
       -> L.ConwayGenesis L.StandardCrypto
+      -- \^ The genesis data to amend
       -> L.ConwayGenesis L.StandardCrypto
     addDRepsToConwayGenesis dRepKeys stakingKeys conwayGenesis =
-      conwayGenesis
-        { L.cgDelegs = delegs (zip stakingKeys (case dRepKeys of [] -> []; _ -> cycle dRepKeys))
-        , L.cgInitialDReps = initialDReps (L.ucppDRepDeposit $ L.cgUpgradePParams conwayGenesis) dRepKeys
-        }
+      conwayGenesis{L.cgDelegs, L.cgInitialDReps}
      where
-      delegs
-        :: [(VerificationKey StakeKey, VerificationKey DRepKey)]
-        -> ListMap (L.Credential L.Staking L.StandardCrypto) (L.Delegatee L.StandardCrypto)
+      -- The credential, to the drep it delegates to
+      delegs :: [(L.Credential L.Staking L.StandardCrypto, VerificationKey DRepKey)]
       delegs =
```

Comment:
Note to @palas: this new `delegs` value is like the old `cgDelegs` one, except that it doesn't hardcode applying some constructors right away in nested positions. This allows to use `delegs` at the source for computing `cgDelegs` and `cgInitialDReps`. This is important to make sure we are consistent between the two.

Reply:
This is a more convenient representation, but I was told that testers need this to be very efficient because they run it with millions of stake pools using the `--transient-*` options. So I benchmarked on my computer the following command:

```bash
cardano-cli latest genesis create-testnet-data --transient-stake-delegators 1000000 --transient-drep-keys 1000000 --out-dir test
```

And, before this commit it took about 36 seconds and used about 1.62 GB of RAM.
But, after the commit, I stopped it at 10 min and had used up to 2.63 GB of RAM by then.

So I think we gotta optimise it. And I am not sure what the cause is, but I think it was was related to either the `Credential` or the `Delegatee` being too large to have as an intermediate representation.